cdap-data-fabric-tests/src/test/java/co/cask/cdap/data2/dataset2/lib/table/hbase/HBaseMetricsTableTest.java

```diff
@@ -86,6 +96,36 @@ public static void tearDown() throws Exception {
   }
 
   @Override
+  @Test
+  public void testConcurrentIncrement() throws Exception {
+    final MetricsTable table = getTable("testConcurrentIncrement");
+    final int rounds = 500;
+    Map<byte[], Long> inc1 = ImmutableMap.of(X, 1L, Y, 2L);
+    Map<byte[], Long> inc2 = ImmutableMap.of(Y, 1L, Z, 2L);
+    // HTable used by HBaseMetricsTable is not thread safe, so each thread must use a separate instance
+    // HBaseMetricsTable does not support mixed increment and incrementAndGet so the
+    // updates and assertions here are different from MetricsTableTest.testConcurrentIncrement()
+    Collection<? extends Thread> threads =
+        ImmutableList.of(new IncThread(getTable("testConcurrentIncrement"), A, inc1, rounds),
+            new IncThread(getTable("testConcurrentIncrement"), A, inc2, rounds),
+            new IncAndGetThread(getTable("testConcurrentIncrement"), A, R, 5, rounds),
+            new IncAndGetThread(getTable("testConcurrentIncrement"), A, R, 2, rounds));
+    for (Thread t : threads) {
+      t.start();
```

Comment:
just occurred to me we don't use any barrier to make sure that threads start at same time (to ensure we actually test concurrent runs)